Tests/UpstreamIntegrationTests/ConfigOverrideTests.swift

```diff
@@ -0,0 +1,327 @@
+//
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+@testable import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPServices
+import Foundation
+import XCTest
+
+/// Performs validation on integration with the Edge Network upstream service
+class ConfigOverrideTests: TestBase {
+    private var edgeEnvironment: EdgeEnvironment = .prod
+    private var edgeLocationHint: EdgeLocationHint?
+
+    private var networkService: RealNetworkService = RealNetworkService()
+
+    let LOG_SOURCE = "ConfigOverrideTests"
+
+    let asyncTimeout: TimeInterval = 10
+
+    let url = "https://obumobile5.data.adobedc.net/ee/v1/interact"
+
+    // Run before each test case
+    override func setUp() {
+        ServiceProvider.shared.networkService = networkService
+
+        super.setUp()
+
+        continueAfterFailure = true
+        TestBase.debugEnabled = true
+        // Extract Edge Network environment level from shell environment; see init for default value
+        self.edgeEnvironment = EdgeEnvironment()
+        print("Using Edge Network environment: \(edgeEnvironment.rawValue)")
+
+        // Extract Edge location hint from shell environment; see init for default value
+        // self.edgeLocationHint = EdgeLocationHint()
+
+        let waitForRegistration = CountDownLatch(1)
+        MobileCore.setLogLevel(.trace)
+
+        // Set environment file ID for specific Edge Network environment
+        MobileCore.configureWith(appId: getTagsEnvironmentFileId(for: edgeEnvironment))
+
+        MobileCore.registerExtensions([Identity.self, Edge.self], {
+            print("Extensions registration is complete")
+            waitForRegistration.countDown()
+        })
+        XCTAssertEqual(DispatchTimeoutResult.success, waitForRegistration.await(timeout: 2))
+
+        // Set Edge location hint value if one is set for the test target
+        if edgeLocationHint != nil {
+            print("Setting Edge location hint to: \(String(describing: edgeLocationHint?.rawValue))")
+            Edge.setLocationHint(edgeLocationHint?.rawValue)
+        } else {
+            print("No preset Edge location hint is being used for this test.")
+        }
```

Comment:
this can be an EdgeHelper too

Reply:
What are you thinking when you say as a helper? I have pushed some updates, but I think I might have not extracted this lcoationHint and edgeEnvironment as per your thinking. Let me know if I missed something in the updates.

Reply:
I was referring to L56 -> L62 which can be extracted in TestBase+EdgeHelpers in a simple func setInitialLocationHint(edgeLocationHint) so all tests can use it as is and also save some lines

Reply:
I did not move that since we don't register Edge extension in the TestBase.